tools/schemas/schema.ts

```diff
@@ -11,4 +11,46 @@ const MonorepoSchema = z.object({
   patternGroups: UrlSchema,
 });
 
-export { MonorepoSchema };
+const PackageRuleSchema = z.object({
+  matchCurrentVersion: z.string().optional(),
+  matchDatasources: z.array(z.string()),
+  matchPackageNames: z.array(z.string()),
+  replacementName: z.string().optional(),
+  replacementVersion: z.string().optional(),
+  description: z.string().optional(),
+  replacementNameTemplate: z.string().optional(),
+});
+
+const RuleSetSchema = z.object({
+  description: z.string(),
+  packageRules: z
+    .array(PackageRuleSchema)
+    .min(1)
+    .refine(
+      (rules) =>
+        rules.some(
+          (rule) =>
+            rule.replacementName !== undefined ||
+            rule.replacementNameTemplate !== undefined,
+        ),
+      {
+        message:
+          'At least one package rule must use either the replacementName config option, or the replacementNameTemplate config option',
```

Comment:
Shouldn't all rules containing a `replacement` name.

Reply:
[There is one which does not.](https://github.com/renovatebot/renovate/blob/1562b303449c68e6529ad66a6924db5bf8a36e2a/lib/data/replacements.json#L183-L191) .

I am not sure it is needed here. But, decided to accomodate for it in in the Schema.